File: apps/api/app/schemas/goal_progress.py
"""Pydantic schemas for goal progress."""

from datetime import date

from pydantic import BaseModel


class GoalSupportingItem(BaseModel):
    id: str
    name: str
    type: str
    adherence_rate: float | None = None
    taken_count: int
    total_count: int


class GoalTrendPoint(BaseModel):
    date: str
    value: int


class GoalProgressItem(BaseModel):
    goal: str
    label: str
    icon: str
    item_count: int
    adherence_rate: float | None = None
    supporting_items: list[GoalSupportingItem]
    journal_metric: str | None = None
    journal_avg: float | None = None
    journal_trend: list[GoalTrendPoint] = []


class GoalProgressResponse(BaseModel):
    goals: list[GoalProgressItem]
    has_preferences: bool
    period_days: int | None = None
    start_date: date | None = None
    end_date: date | None = None

File: packages/domain/src/goal-progress.ts
export interface GoalSupportingItem {
  id: string;
  name: string;
  type: string;
  adherence_rate: number | null;
  taken_count: number;
  total_count: number;
}

export interface GoalTrendPoint {
  date: string;
  value: number;
}

export interface GoalProgressItem {
  goal: string;
  label: string;
  icon: string;
  item_count: number;
  adherence_rate: number | null;
  supporting_items: GoalSupportingItem[];
  journal_metric: string | null;
  journal_avg: number | null;
  journal_trend: GoalTrendPoint[];
}

export interface GoalProgressResponse {
  goals: GoalProgressItem[];
  has_preferences: boolean;
  period_days: number | null;
  start_date: string | null;
  end_date: string | null;
}
